src/dialog.js

```diff
@@ -449,6 +457,112 @@ genotet.dialog.upload_ = function() {
 };
 
 /**
+ * Creates a dialog for signing up.
+ * @private
+ */
+genotet.dialog.signUp_ = function() {
+  var modal = $('#dialog');
+  modal.find('.modal-content').load(genotet.dialog.TEMPLATES_.signUp,
+    function() {
+      modal.modal();
+      var email = modal.find('#email');
+      var username = modal.find('#username');
+      var password = modal.find('#password');
+      var confirmPassword = modal.find('#confirm-password');
+      var btnSignUp = modal.find('#btn-sign-up').prop('disabled', true);
+
+      // Checks if all required fields are filled.
+      var uploadReady = function() {
+        return email.val() && username.val() && password.val() &&
+          (password.val() == confirmPassword.val());
+      };
+      confirmPassword.on('input', function() {
+        btnSignUp.prop('disabled', !uploadReady());
+      });
+      btnSignUp.click(function() {
+        var userInfo = {
+          type: 'sign-up',
+          email: email.val(),
+          username: username.val(),
+          password: CryptoJS.SHA256(password.val()).toString(),
+          confirmed: false
+        };
+
+        $.ajax({
+          url: genotet.data.userURL,
+          type: 'POST',
+          data: userInfo,
+          dataType: 'json'
+        }).done(function(data) {
+            if (!data.success) {
+              genotet.error('failed to signed up', data.message);
+            } else if (data.response.success) {
+              genotet.menu.displaySignedUser(userInfo.username);
+              genotet.data.userInfo.username = userInfo.username;
+              genotet.success('signed up');
+            } else {
+              genotet.error(data.response.errorMessage);
+            }
+          })
+          .fail(function(res) {
+            genotet.error('failed to signed up');
+          });
+      });
+    });
+};
+
+/**
+ * Creates a dialog for signing in.
+ * @private
+ */
+genotet.dialog.signIn_ = function() {
+  var modal = $('#dialog');
+  modal.find('.modal-content').load(genotet.dialog.TEMPLATES_.signIn,
+    function() {
+      modal.modal();
+      var username = modal.find('#username');
+      var password = modal.find('#password');
+      var btnSignIn = modal.find('#btn-sign-in').prop('disabled', true);
+
+      // Checks if all required fields are filled.
+      var uploadReady = function() {
+        return username.val() && password.val();
+      };
+      password.on('input', function() {
+        btnSignIn.prop('disabled', !uploadReady());
+      });
+      btnSignIn.click(function() {
+        var userInfo = {
+          type: 'sign-in',
+          username: username.val(),
+          password: CryptoJS.SHA256(password.val()).toString()
```

Comment:
Check password strength. For the simpliest it should contain at least 8 characters.